tests/ut/codegen/test_external_kernel.py

```
@@ -138,7 +139,7 @@ def test_all_external_graph_emits_manifest_when_ptoas_is_skipped(tmp_path):
 def test_external_source_requires_aic_or_aiv(tmp_path):
     """external_source on a non-AIC/AIV function is rejected with a clear error."""
     cpp = _write_kernel(tmp_path)
-    with pytest.raises(Exception, match="external_source is only valid on FunctionType.AIC"):
+    with pytest.raises(ParserTypeError, match="external_source is only valid on FunctionType.AIC"):
```

Ruff (0.16.0)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)